app/modules/feeds-and-posts.js

```diff
@@ -0,0 +1,57 @@
+import { Promise } from 'rsvp';
+
+let fpPromise = null;
+
+/**
+ * Gets the Feeds & Posts ES Module, by first appending the F&P library to the page
+ *
+ * @returns {Promise} Promise of module
+ */
+function getModule() {
+	if (fpPromise) {
+		return fpPromise;
+	}
+
+	fpPromise = new Promise((resolve, reject) => {
+		const fpScript = document.createElement('script');
+		fpScript.onload = () => {
+			resolve(window.fp);
```

Comment:
there are internet providers that will wrap your script in their script loading mechanism - introducing here a race condition - does not happen super often, so I'd suggest doing a check here if window.fp is available and if not - reject promise